### PR TITLE
Prevents rollerbeds and cocoons from being dragged by xenos

### PR DESCRIPTION
**Pull request by lKiseki**
## About The Pull Request
Per title.
The way I approached this code-wise was make a new status flag, `CANNOTDRAG`, then applied it to rollerbeds and cocoons. It is now as simple as adding this status flags to any other atoms that shouldn't be dragged.

## Why It's Good For The Game
Though I definitely disagree with the existence of a blacklist/whitelist for what can and cannot be dragged, I don't think rollerbeds and cocoons should remain something xenos can drag. You already cannot drag people that're dead, so why should we let xenos drag people that are dead but stuck in a cocoon, or on a rollerbed? It's an unnecessary form of punishment for unsuspecting marines trying to help someone else out.

## Changelog
:cl: Lewdcifer
balance: Rollerbeds and cocoons can no longer be dragged by xenos.
code: Added a new status flag, CANNOTDRAG. This can be applied to any atom that shouldn't be dragged by xenos.
/:cl:

### CODE REVIEW

**Comment by Strayed-Raven** [comment]
Countdown to being added to released but unopen cocoons.

**Comment by FosterGrime-Heymynameslouie** [comment]
xenos can also push the roller beds with people buckled on them too?

**Comment by lKiseki** [comment]
> xenos can also push the roller beds with people buckled on them too?

That's going to be an entirely different situation that I'd deem out of the scope of this PR, as much as I dislike the existence of such a feature.

**Comment by lKiseki** [comment]
> Countdown to being added to released but unopen cocoons.

Good reminder. Let me add that to this PR.

**Comment by XElectricX** [comment]
Weirdest salt PR so far. Who asked for this.

**Comment by lovegreenstuff** [comment]
> Weirdest salt PR so far. Who asked for this.

i did.
what are you going to do about it, coward?

**Comment by ThelastArchiver** [comment]
Up until now the cocoon dragging was a good way to punish marines for trying to secure a cocoon without having control over the area but I have to admit that brazilling a cocoon is unfun for the dead person so how about making it so that aliens only get the points if the cocoon is nested or add a 10 sec delay for aliens for grabing a cocoon

**Comment by EvilMelonTheSecond** [comment]
How about stasis bags? They are the same as rollerbeds since xeno can slash them open in the cave corner

**Comment by Losenis** [comment]
Blacklisting, called it. Called it since the first day of xenodrag.

**Comment by SilencedMP5A5** [comment]
benos can still buckle people to rollerbeds tho

**Comment by 567Turtle** [comment]
> benos can still buckle people to rollerbeds tho

thats funny and doesnt end with you almost 100% going perma, getting taken 200 meters from the frontlines while getting brought back to FoB isnt funny and does end with going perma

**Comment by novaepee** [comment]
I for one can't wait for when marine contributors slap CANNOTDRAG for anything else that they don't want dragged by xenomorphs if this PR gets merged.

**Comment by Loldaddy5676** [comment]
> 

thing is you have to rely on a marine to remove the cacoon from its nest however xenos can also just melt the cacoon? if this is gonna get merged atleast fix the issue of xenos being able to melt their own strucutres

**Comment by Losenis** [comment]
>i feel like if a xeno is able to recover cocoons from the marines they should be able to drag it back to safety for the sweet points

The whole point of cocooning is bigger reward for that specific risk of marines nabbing the cocoon compared to psydraining on the spot.

Also I don't see xenos melting cocoons being an issue given melting now ejects contents, they're only shooting themselves in the foot. Or are we talking about the nest that contains the cocoon to free it?

**Comment by Loldaddy5676** [comment]
> > i feel like if a xeno is able to recover cocoons from the marines they should be able to drag it back to safety for the sweet points
> 
> The whole point of cocooning is bigger reward for that specific risk of marines nabbing the cocoon compared to psydraining on the spot.
> 
> Also I don't see xenos melting cocoons being an issue given melting now ejects contents, they're only shooting themselves in the foot. Or are we talking about the nest that contains the cocoon to free it?

the nest itself which might be the problem.

**Comment by Losenis** [comment]
Very fair point.

**Comment by FIXTHERADBOXPLEASE** [comment]
Being able to push rollerbeds is already plenty, dragging just leads to stupid shit if the marine's Medic makes a mistake. Cocoons being draggable was obviously a mistake. All my comments for this PR.

**Comment by chizzy376** [comment]
Which cocoons does this effect, exactly? The ones that are prespawned in every map, or the ones made by xenos on psy cocoon? Or both? For the latter, I wholeheartedly agree, they shouldn't be moved. But for the former, I need to be able to move them as right now they disallow building on them so you need to slash them and move them.

**Comment by Loldaddy5676** [comment]
i mean its still a marine skill issue, the cocoon part. not allowing xenos to use rollers beds is fine but a marine HAS to release the cocoon off the nest. this goes with literally anything else. plasma cutters, special weapons, etc. if you lose a asset then you should be punished. sure it might be unfun for the marine inside the cocoon however they should not have A) run off alone and B) if they do want to chase a xeno, to bring a friend with them to drag them away from being cocooned. 
tldr: roller beds sure, skill issue losing a cocoon you just released.

**Comment by Losenis** [comment]
>this goes with literally anything else

"Literally anything else", including cocoons, were introduced at a time where xenos could not drag and could only acid on the spot.

**Comment by Lumipharon** [comment]
I'm pretty against making a blacklist as it becomes very arbitrary. I don't really like xenos being able to pull in general though.

**Comment by Loldaddy5676** [comment]
> > this goes with literally anything else
> 
> "Literally anything else", including cocoons, were introduced at a time where xenos could not drag and could only acid on the spot.

well yeah, if the xenos dont play their cards right they can lose their buildings. defenses. and cocoons. same goes with marines. if they lose a recovered cocoon, its their fault. the game imo is all about whoever gets to hold/ take back their assets.

**Comment by Losenis** [comment]
The argument doesn't apply due to none of the aforementioned things being introduced with xeno dragging in mind. Cocoons were not introduced with xeno dragging in mind, because if they had been then destroying the nest would've just automatically ejected the marine rather than leaving a cocoon that, at the time of introduction, xenos were deliberatedly unable to drag much like a lot of other things.

**Comment by QualityVan** [comment]
Xenos can't buckle corpses to rollerbeds themselves, so any time that happens it's strictly a marine's fault. Can they detach cocoons, or is that also only from marine actions?

**Comment by Loldaddy5676** [comment]
> The argument doesn't apply due to none of the aforementioned things being introduced with xeno dragging in mind. Cocoons were not introduced with xeno dragging in mind, because if they had been then destroying the nest would've just automatically ejected the marine rather than leaving a cocoon that, at the time of introduction, xenos were deliberatedly unable to drag much like a lot of other things.

ah misunderstood you in the earlier post. k then

**Comment by Loldaddy5676** [comment]
> Xenos can't buckle corpses to rollerbeds themselves, so any time that happens it's strictly a marine's fault. Can they detach cocoons, or is that also only from marine actions?

as far as i know it. only marines can relase the cocoons

**Comment by Losenis** [comment]
>Xenos can't buckle corpses to rollerbeds themselves, so any time that happens it's strictly a marine's fault.
Two problems here:

First, xenos pushing rollerbed buckling is an exception because that happened before xenodrag by just pulling the marine, but that had the limitation of only being pushable instead of draggable; if they pushed the buckled marine into a corned, that's where it stopped. In most maps it's almost a guarantee that the average xeno will hit a corner sooner or later, but with pulling this risk is entirely removed and you can drag a buckled roller to the silo without risk.

Second, it's not the fault of the marine who got buckled, it's the fault of the marine that is doing the buckling. The marine in question is punished for the skill issues of another marine who is not punished whatsoever for said skill issue unless he's first killed off, at which point a wipe is probably happening because if xenos can get to someone running away with a rollerbed, the front's probably dead. The only skill issue the buckled marine went through is not preemptively checking for baldness in the corpsman and killing him before he can buckle for xenos to drag. Numerous things got removed specifically due to this "punish another for your own skill issue while nothing bad happens to you" ideas, like defender's bullet reflection, russian red's massive delayed stun, and the whole mess that had all marines get a mild natural resistance to FF from ages ago.
